# 'die' is not available until GHC 7.10
--- testsuite/timeout/timeout.hs
+++ testsuite/timeout/timeout.hs
@@ -30,8 +30,8 @@ main = do
       [secs,cmd] ->
           case reads secs of
           [(secs', "")] -> run secs' cmd
-          _ -> die ("Can't parse " ++ show secs ++ " as a number of seconds")
-      _ -> die ("Bad arguments " ++ show args)
+          _ -> ((>> exitFailure) . hPutStrLn stderr) ("Can't parse " ++ show secs ++ " as a number of seconds")
+      _ -> ((>> exitFailure) . hPutStrLn stderr) ("Bad arguments " ++ show args)
 
 timeoutMsg :: String
 timeoutMsg = "Timeout happened...killing process..."
